USWDS - Date range picker: Add props controls and stories

amyleadem opened the pull request:
<!-- Please feel free to remove whatever sections/lines in this aren’t relevant.

Use the title line as the title of your pull request, then delete these lines.

## Title line template: [Title]: Brief description

Website: For pull requests that impact designsystem.digital.gov’s look, feel, or functionality, please open a pull request on the uswds-site repo (https://github.com/uswds/uswds-site).

-->

## Description
Part 2/2, closes https://github.com/uswds/uswds/issues/4247

Preview link: [Date range picker component stories](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/al-date-range-picker/?path=/story/components-form-inputs-date-range-picker--default)

### Problem 
Our documentation provided incorrect information about where to add `data-min-date` and `data-max-date` attributes to the date range picker component. Additionally, we do not have have stories to demonstrate how these properties work on date range picker. 

### Two-part solution
This PR is part 2 of a two-step solution to close https://github.com/uswds/uswds/issues/4247. 

- Part 1: USWDS-Site - Update component instructions to point to the correct attribute elements
    - Part 1 can be found in this PR: https://github.com/uswds/uswds-site/pull/1605
- Part 2: USWDS - Create stories for the date range picker component to demonstrate correct placement and behavior of the component's properties.

## Additional information


Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [ ] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

mejiaj reviewed (commented):
Thanks for outlining the solution!

I haven't been able to update these values locally. Video attached.
https://user-images.githubusercontent.com/3385219/171697522-76f05c2c-0275-4dac-8b5f-a156c97d1d1f.mp4

**Steps to reproduce**
1. Load page and ensure component is initialized (icons are showing)
2. Click on `minDate` input field
3. Select date from calendar popup; I chose 06/03/22
  ![image](https://user-images.githubusercontent.com/3385219/171697829-0ef2bc24-b0eb-4c0e-9f1b-6469d2d2d4bc.png)
4. Verify calendar icons disappear, but no error appears
  ![image](https://user-images.githubusercontent.com/3385219/171697939-c6648577-11d8-4f7b-8a62-8594b6df57be.png)
5. Click inside **Event start date** input
6. View error appear in console
7. Click **Event end date** input and verify another error

The error:
```console
Uncaught TypeError: can't access property "value", n is null
    Q index.js:658
    s index.js:1252
    f index.js:2200
    exports index.js:7
    exports index.js:4
    exports index.js:3
    add index.js:60
    add index.js:59
    r behavior.js:14
    r behavior.js:13
    ./dist/js/uswds.min.js/<[41]</</< start.js:21
    ./dist/js/uswds.min.js/<[41]</< start.js:20
```

Points to `usa-date-picker/src/index.js:658` 
```js
  const internalInputEl = datePickerEl.querySelector(
    DATE_PICKER_INTERNAL_INPUT
  );
```

amyleadem commented on packages/usa-date-range-picker/src/usa-date-range-picker.json:
```diff
@@ -0,0 +1,7 @@
+{
+  "defaultValueStart": "",
+  "defaultValueEnd": "",
```
@mejiaj I have updated the controls used to be `text` instead of `date`, since that is what the attribute was really looking for. I also updated the variable and control names to be a bit more intuitive. Please let me know if you have feedback!

amyleadem commented:
@mejiaj I have made a few updates: 
- Updated the controls used to be `text` instead of `date`, since that is what the attribute was really looking for. 
- Updated the variable and control names to be a bit more intuitive
- Selectively showed/hid controls on the variant stories to better isolate the demos
Please let me know if you have feedback!

mejiaj reviewed (approved):
Looks good, thanks Amy!

The controls make sense and look good. There's a minor issue in Edge/Chromium where the component calendar disappears when modifying the controls.

![image](https://user-images.githubusercontent.com/3385219/174343779-ab9d8eca-c966-466d-8da3-5bebbf8b0039.png)

It then comes back on refresh with the right value
![image](https://user-images.githubusercontent.com/3385219/174343866-ecbb8d42-4f49-4fd1-ac1e-daf7e931e1b2.png)

Just adding this for reference, don't think there's anything we can do about it at the moment.
